pkg/batchpertrace/go.mod

```diff
@@ -0,0 +1,8 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpertrace
```

Comment:
We do not have a convention for a directory for libraries. `pkg` seems ok. @bogdandrutu @jrcamp any thoughts?

Reply:
Is the intention for something outside of opentelemetry-collector-contrib to use it? Having libraries in contrib seems odd. If just code in contrib is going to use it we've been putting things in internal.

Reply:
It is also for usage by external components: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1257#discussion_r504862655

Reply:
ping @jrcamp , @tigrannajaryan

Reply:
sgtm